rust/datafusion/src/sql/parser.rs

```diff
@@ -77,20 +77,20 @@ pub struct DFParser {
 
 impl DFParser {
     /// Parse the specified tokens
-    pub fn new(sql: &str) -> Result<Self, ParserError> {
-        let dialect = GenericDialect {};
-        let mut tokenizer = Tokenizer::new(&dialect, sql);
+    pub fn new(sql: &str, dialect: &dyn Dialect) -> Result<Self, ParserError> {
+        let mut tokenizer = Tokenizer::new(dialect, sql);
         let tokens = tokenizer.tokenize()?;
         Ok(DFParser {
             parser: Parser::new(tokens),
         })
     }
 
     /// Parse a SQL statement and produce a set of statements
-    pub fn parse_sql(sql: &str) -> Result<Vec<Statement>, ParserError> {
-        let mut tokenizer = Tokenizer::new(&GenericDialect {}, &sql);
-        tokenizer.tokenize()?;
-        let mut parser = DFParser::new(sql)?;
+    pub fn parse_sql(
+        sql: &str,
+        dialect: &dyn Dialect,
```

Comment:
Perhaps we should store the dialect in the DFParser struct so we don't have to pass it in here as well?

Reply:
I think dialect is an important optional parameter and will only be used when initializing, so I think it should be entered in the entry

Reply:
Actually, I had misread this. I had thought that `parse_sql` was a method that took `&self` but it does not, so I think this is fine.